Tools/Scripts/webkitpy/common/find_files.py

```diff
@@ -79,7 +91,8 @@ def sort_by_directory_key(files_list):
 
         return sorted(files_list, key=directory_sort_key)
 
-    paths_to_walk = itertools.chain(*(sort_by_directory_key(filesystem.glob(path)) for path in paths))
-
-    all_files = itertools.chain(*(sort_by_directory_key(filesystem.files_under(path, skipped_directories, file_filter)) for path in paths_to_walk))
+    paths_to_walk = itertools.chain(*(sort_by_directory_key(_filter_test_variant(path, filesystem.glob)) for path in paths))
+    all_files = itertools.chain(*(sort_by_directory_key(
+        _filter_test_variant(path, lambda p: filesystem.files_under(p, skipped_directories, file_filter))
+    ) for path in paths_to_walk))
```

Comment:
Suggested refactor:

```python3
def sorted_paths_generator(fs_selector):
    for test_selector in paths_to_walk:
        path, separator, variant = test_selector.partition('?')
        matching_tests = (matching_path + separator + variant for matching_path in fs_selector(path))
        if directory_sort_key:
            yield from sorted(matching_tests, key=directory_sort_key)
        else:
            yield from matching_tests

paths_to_walk = sorted_paths_generator(filesystem.glob)
all_files = sorted_paths_generator(lambda p: filesystem.files_under(p, skipped_directories, file_filter))
```

This would replace `_filter_test_variant` and the local `sort_by_directory_key` function altogether. Using `str.partition` instead of `str.split` removes a bunch of the conditional logic.